go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -509,17 +528,22 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 	lsp *nbdb.LogicalSwitchPort, podAnnotation *util.PodAnnotation, newlyCreatedPort bool, err error) {
 	var ls *nbdb.LogicalSwitch
 
-	podDesc := fmt.Sprintf("%s/%s/%s", nadName, pod.Namespace, pod.Name)
-	switchName, err := bnc.getExpectedSwitchName(pod)
+	if err := bnc.ensureNetworkInfoForVM(pod); err != nil {
```

Comment:
Later on when is neede the pod is retrieved again from the informers cache (it was done like that originaly) we cannot change the pod we are passing since it's the one at the informers cache (in fact some unit test fails with the race condition checker)

Reply:
What do you mean by `later on`? When\Where is the pod retrieved again?

Reply:
At `updatePodAnnotationWithRetry` they read again from informers cache.

Reply:
`updatePodAnnotationWithRetry` it's at the end of `addLogicalPortToNetwork` so that's too late and I am not sure how that's relevant. 

I am talking about how can the remainder of `addLogicalPortToNetwork` work if `ensureNetworkInfoForVM` does not provide any of the information it has copied from the original pod?

Reply:
Am I understanding that right? Isn't that super nasty? How do you know the informer cache will not at some point allocate new memory for the updated pod?

Why don't you just return the information that you need (original switch and original pod annotation) from `ensureNetworkInfoForVM`? And then you don't need to wait for anything.